packages/mobile/src/harmony-native/components/input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,311 @@
+      helperText,
+      startAdornmentText,
+      endAdornmentText,
+      startIcon: StartIcon,
+      endIcon: EndIcon,
+      endAdornment: endAdornmentProp,
+      _incorrectError,
+      _isFocused,
+      _disablePointerEvents,
+      ...other
+    } = props
+
+    const innerInputRef = useRef<RNTextInput>(null)
+
+    const { typography, color, motion } = useTheme()
+
+    let endAdornment: null | ReactNode
+    if (EndIcon != null) {
+      endAdornment = <EndIcon size='m' color='subdued' />
+    } else if (endAdornmentProp != null) {
+      endAdornment = endAdornmentProp
+    } else {
+      endAdornment = null
+    }
+
+    const focused = useSharedValue(_isFocused ? 1 : 0)
+
+    const handleFocus = useCallback(
+      (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        focused.value = withTiming(1, motion.press)
+        onFocusProp?.(e)
+      },
+      [focused, motion.press, onFocusProp]
+    )
+
+    const handleBlur = useCallback(
+      (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
+        focused.value = withTiming(0, motion.press)
+        onBlurProp?.(e)
+      },
+      [focused, motion.press, onBlurProp]
+    )
+
+    const [isFocusedState, onFocus, onBlur] = useFocusState(
+      handleFocus,
+      handleBlur
+    )
+
+    const isFocused = _isFocused ?? isFocusedState
+
+    // For focus behavior and accessiblity, <label> needs to have a htmlFor={} provided to an id matching the input
+    // const backupId = useId()
+    const id = idProp // ?? backupId
+
+    const characterCount = value !== undefined ? `${value}`.length : 0
+    const hasValue = characterCount > 0
+
+    // Hide the label when requested or when the size is set to small
+    const shouldShowLabel = !hideLabel && size !== TextInputSize.SMALL
+    const labelText = required ? `${labelProp} *` : labelProp
+    const placeholderText =
+      required && hideLabel ? `${placeholder} *` : placeholder
+    const helperTextSize: TextSize = size === TextInputSize.SMALL ? 'xs' : 's'
+
+    // Whenever a label isn't visible the placeholder should be visible in it's place (if provided)
+    const shouldShowPlaceholder = isFocused || !shouldShowLabel
+    const shouldShowAdornments = isFocused || hasValue || !shouldShowLabel
+    // Show the maxlength text whenever we hit a certain threshold (default 70%) + the input is focused
+    const shouldShowMaxLengthText =
+      isFocused &&
+      maxLength &&
+      characterCount >= showMaxLengthThreshold * maxLength
+    // Turn the maxlength text to the warning color whenever we hit a certain threshold (default 90%)
+    let maxLengthTextColor: TextColors = 'default'
+    if (maxLength && characterCount > maxLength) {
+      maxLengthTextColor = 'danger'
+    } else if (
+      maxLength &&
+      characterCount >= maxLengthWarningThreshold * maxLength
+    ) {
+      maxLengthTextColor = 'warning'
+    }
+
+    const isSmall = size === TextInputSize.SMALL
+    const stateColor = disabled
```

Comment:
nit: Very small diff but I prefer the word status here personally 
```suggestion
    const statusColor = disabled
```

Reply:
sgtm